Ripped from Fedora

* Wed Dec 15 2004 Tim Waugh <twaugh@redhat.com>
- Call setlocale() so that readline works correctly (bug #142265).

https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=142265

--- netkit-ftp-0.17/ftp/main.c
+++ netkit-ftp-0.17/ftp/main.c
@@ -51,6 +51,7 @@
 
 /* #include <arpa/ftp.h>	<--- unused? */
 
+#include <locale.h>
 #include <signal.h>
 #include <unistd.h>
 #include <string.h>
@@ -109,6 +110,7 @@
 
 	tick = 0;
 
+	setlocale (LC_ALL, "");
 	sp = getservbyname("ftp", "tcp");
 	if (sp == 0) {
 		fprintf(stderr, "ftp: ftp/tcp: unknown service\n");
